js_modules/dagit/packages/core/src/workspace/asset-graph/LaunchAssetChoosePartitionsDialog.tsx

```diff
@@ -259,6 +272,30 @@ export const LaunchAssetChoosePartitionsDialog: React.FC<{
             </Box>
           ) : undefined}
         </Box>
+        {upstreamUnavailableSpans.length > 0 && (
+          <Box margin={{top: 16}}>
+            <Alert
+              intent="warning"
+              title="Upstream Data Missing"
+              description={
+                <>
+                  {upstreamUnavailableSpans
+                    .map((a) =>
+                      a.startIdx === a.endIdx
+                        ? `${selected[a.startIdx]}`
+                        : `${selected[a.startIdx]} through ${selected[a.endIdx]}`,
+                    )
+                    .join(', ')}
```

Comment:
One small nit--when multiple segments of upstream partitions have been materialized, this string will render as "2021-05-05 through 2022-01-18, 2022-01-22 through 2022-01-24 cannot be materialized". I think when there are multiple segments, we can instead display array spans e.g. "Partitions [2021-05-05...2022-01-18], [2022-01-22...2022-01-24] cannot be materialized"

Reply:
Oh I like that! That seems much better because you could copy-paste it easily